src/CaptureClient/CaptureClient.cpp

```diff
@@ -265,16 +262,16 @@ bool CaptureClient::StopCapture() {
   {
     absl::MutexLock lock(&state_mutex_);
     if (state_ == State::kStarting) {
-      // Block and wait until the state is not kStarting
-      bool (*is_not_starting)(State*) = [](State* state) { return *state != State::kStarting; };
-      state_mutex_.Await(absl::Condition(is_not_starting, &state_));
+      LOG("StopCapture ignored, because it is starting and cannot be stopped at this stage.");
+      return false;
     }
 
     if (state_ != State::kStarted) {
```

Comment:
This condition now includes the one above and does basically the same, just with a different string. I would consider listing each state explicitly (`kStarting`, `kStopping`, `kStopped`) and printing a string dedicated to each ("it is starting and cannot be stopped", "it is already stopping", "it is already stopped").

Reply:
done.